Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,315 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsStrArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            CultureInfo.CurrentCulture = GetCustomInvariantCulture();
+
+            decimal max = decimal.MinValue;
+            string maxResult = "";
+            string peopleWithSameHighBalance = "";
+            bool moreThanOne = false;
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                string[] auxBalances = peopleAndBalances[i].Split(',');
+                decimal[] balances = GetBalances(auxBalances[1..]);
+                decimal currentMax = MaxBalance(balances);
+                if (max < currentMax )
+                {
+                    max = currentMax;
+                    peopleWithSameHighBalance = auxBalances[0];
+                    moreThanOne = false;
+                    maxResult = $" had the most money ever. {max:C0}.";
+                }
+               else if (max.Equals(currentMax))
+                {
+                    peopleWithSameHighBalance += $", {auxBalances[0]}";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5069037)